### PR TITLE
Add print statements for the `storage_service_server`

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

The `storage_service_server` did not have logs.

## Proposal

Some print statements are added for tracing.
In passing, it was found that the `read_value_bytes` was not tested and so relevant functions are added.

## Test Plan

The CI is improved by this work.

## Release Plan

Not relevant.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

**Comment by Twey** [comment]
I don't think it's necessary to use both `tracing::instrument` _and_ manually call `tracing::info` at the start of functions. We can enable printing when entering instrumented functions using [`with_span_events`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_span_events) — though I think we should give them the `trace` level.

File: linera-storage-service/src/server.rs [hunk 1]
```diff
@@ -224,11 +225,14 @@ enum ServiceStoreServerOptions {
 
 #[tonic::async_trait]
 impl StoreProcessor for ServiceStoreServer {
+    #[allow(clippy::blocks_in_conditions)]
```
Comment (ma2bd) [inline linera-storage-service/src/server.rs:228]: Should we add `#![allow(clippy::blocks_in_conditions)]` at the beginning of the file instead?

File: linera-storage-service/src/server.rs [hunk 2]
```diff
@@ -508,6 +558,7 @@ async fn main() {
         pending_big_reads,
     };
     let endpoint = endpoint.parse().unwrap();
+    info!("Start of storage_service_service on endpoint={}", endpoint);
```
Comment (ma2bd) [inline linera-storage-service/src/server.rs:561]: "Start of" -> "Starting"

**Comment by ma2bd** [comment]
> I don't think it's necessary to use both `tracing::instrument` _and_ manually call `tracing::info` at the start of functions. We can enable printing when entering instrumented functions using [`with_span_events`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_span_events) — though I think we should give them the `trace` level.

Indeed, both seem to have the same default logging level: https://docs.rs/tracing/latest/tracing/attr.instrument.html

File: linera-storage-service/src/server.rs (hunk 2 above)
Comment (MathieuDutSik) [reply linera-storage-service/src/server.rs]: Done.

File: linera-storage-service/src/server.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-storage-service/src/server.rs]: Done.

File: linera-views/src/test_utils/mod.rs [hunk 3]
```diff
@@ -281,7 +281,12 @@ pub async fn run_reads<S: LocalKeyValueStore>(store: S, key_values: Vec<(Vec<u8>
         }
         assert_eq!(set_key_value1, set_key_value2);
     }
-    // Now checking the read_multi_values_bytes
+    // Now checking the `read_value_bytes`
+    for (key, value) in &key_values {
+        let value_read = store.read_value_bytes(key).await.unwrap();
+        assert_eq!(value_read, Some(value.clone()));
+    }
+    // Now checking the `read_multi_values_bytes`
```
Comment (andresilva91) [inline linera-views/src/test_utils/mod.rs:289]: This seems unrelated to this PR, wouldn't it be better to have a separate PR for it?

**Review by andresilva91 — CHANGES_REQUESTED** [review]
Back to you for the changes suggested by @ma2bd and @Twey

File: linera-views/src/test_utils/mod.rs (hunk 3 above)
Comment (MathieuDutSik) [reply linera-views/src/test_utils/mod.rs]: Ok, I put it in the multi read PR for ScyllaDb where it is a little better.

File: linera-storage-service/src/server.rs [hunk 4]
```diff
@@ -477,6 +498,15 @@ impl StoreProcessor for ServiceStoreServer {
 
 #[tokio::main]
 async fn main() {
+    let env_filter = tracing_subscriber::EnvFilter::builder()
+        .with_default_directive(tracing_subscriber::filter::LevelFilter::INFO.into())
+        .from_env_lossy();
+    tracing_subscriber::fmt()
+        .with_span_events(FmtSpan::NEW)
```
Comment (Twey) [inline linera-storage-service/src/server.rs:505]: I think it would be really nice to mimic [what `test-log` does here](https://github.com/d-e-s-o/test-log/blob/756d49ee2b96cc93f332549e5173017bebb6be58/macros/src/lib.rs#L202-L224) and support runtime configuration by the `RUST_LOG_SPAN_EVENTS` environment variable.

Comment (MathieuDutSik) [reply linera-storage-service/src/server.rs]: Ok, I implemented that.

**Review by Twey — APPROVED** [review]
Looks good to me!  Thanks for this, I'm sure I'll appreciate it in the future :)

**Review by afck — APPROVED** [review]
